### PR TITLE
ATV-62 | Add staff checks for attachments

### PR DESCRIPTION
**Pull request by a-rmz**
## Description :sparkles:
* Add staff checks for attachments

### Closes :no_good_woman:
**[ATV-62](https://helsinkisolutionoffice.atlassian.net/browse/ATV-62):** Add the permission checking for Staff users

### Automated tests :gear:️
```shell
$ pytest documents/tests/test_api_attachment_detail.py
```

### CODE REVIEW

File: documents/api/viewsets.py [hunk 1]
```diff
@@ -69,8 +69,9 @@ def get_queryset(self):
 
         return Attachment.objects.filter(**qs_filters)
 
-    def retrieve(self, request, *args, **kwargs):
-        raise MethodNotAllowed(request.method)
+    @staff_required(required_permission=ServicePermissions.VIEW_DOCUMENTS)
```
Comment (charn) [inline documents/api/viewsets.py:72]: Should we use `VIEW_ATTACHMENTS` permission here instead?

File: documents/tests/test_api_attachment_detail.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+from atv.tests.factories import GroupFactory
+from documents.tests.factories import AttachmentFactory
+from services.enums import ServicePermissions
+
+
+def test_attachment_detail_service_user(api_client, user, attachment):
+    group = GroupFactory(name=attachment.document.service.name)
+    user.groups.add(group)
+    assign_perm(
+        ServicePermissions.VIEW_DOCUMENTS.value, group, attachment.document.service
```
Comment (charn) [inline documents/tests/test_api_attachment_detail.py:14]: Or `VIEW_ATTACHMENTS`?

**Review by charn — APPROVED** [review]
Made a note about the permission which is used on the attachment view. LGTM! :+1: :shipit:

File: documents/api/viewsets.py (hunk 1 above)
Comment (a-rmz) [reply documents/api/viewsets.py:72]: Good catch!

File: documents/tests/test_api_attachment_detail.py (hunk 2 above)
Comment (a-rmz) [reply documents/tests/test_api_attachment_detail.py:14]: Same hehe

**Pull request merged**